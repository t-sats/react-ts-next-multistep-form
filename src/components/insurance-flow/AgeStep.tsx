import React, { useState } from "react";
import { useFormState } from "@/context/FormContext";
import { validateAge } from "@/helpers/validation";

const AgeStep = () => {

  const [error, setError] = useState("");

  const { formData, handleChange, handleNext, handleBack }  = useFormState();

  const ageValue = formData.age;

  const handleOnFocus = () => {
      setError("");
  };

  const handleSubmit = () => {
      if (!validateAge(ageValue)) {
          setError("Please enter valid age");
      } else {
          handleNext();
      };
  };

  return (
    <div className="flex flex-col gap-4 w-full">
      <div className="flex flex-col gap-2 items-start">
        <label htmlFor="age">Age:</label>
        <input
          id="age"
          type="number"
          onChange={handleChange}
          onFocus={handleOnFocus}
          name="age"
          value={formData.age}
          className="input input-primary w-full max-w-xs"
        />
          {error}
      </div>
      <button
        className="rounded-full border border-solid border-transparent bg-foreground text-background gap-2 hover:bg-[#28a745] font-medium text-sm sm:text-base h-10 sm:h-12 px-4 sm:px-5 sm:w-auto"
        onClick={handleBack}
        type="button"
      >
        Back
      </button>
      <button
        type="button"
        className="rounded-full border border-solid border-transparent bg-foreground text-background gap-2 hover:bg-[#28a745] font-medium text-sm sm:text-base h-10 sm:h-12 px-4 sm:px-5 sm:w-auto"
        onClick={handleSubmit}
      >
        Buy Now
      </button>
    </div>
  );
};

export default AgeStep;
