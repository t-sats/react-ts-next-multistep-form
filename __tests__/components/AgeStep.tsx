import React from 'react';
import { render } from '@testing-library/react';
import AgeStep from '@/components/insurance-flow/AgeStep';

it('renders AgeStep component correctly', () => {
  const { container } = render(<AgeStep />)

  expect(container).toMatchSnapshot()
});
