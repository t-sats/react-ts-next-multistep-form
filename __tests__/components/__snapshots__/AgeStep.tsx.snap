// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AgeStep component correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-4 w-full"
  >
    <div
      class="flex flex-col gap-2 items-start"
    >
      <label
        for="age"
      >
        Age:
      </label>
      <input
        class="input input-primary w-full max-w-xs"
        id="age"
        name="age"
        type="number"
        value="1"
      />
    </div>
    <button
      class="rounded-full border border-solid border-transparent bg-foreground text-background gap-2 hover:bg-[#28a745] font-medium text-sm sm:text-base h-10 sm:h-12 px-4 sm:px-5 sm:w-auto"
      type="button"
    >
      Back
    </button>
    <button
      class="rounded-full border border-solid border-transparent bg-foreground text-background gap-2 hover:bg-[#28a745] font-medium text-sm sm:text-base h-10 sm:h-12 px-4 sm:px-5 sm:w-auto"
      type="button"
    >
      Buy Now
    </button>
  </div>
</div>
`;
